tendrl/commons/flows/create_cluster/__init__.py

```diff
@@ -50,119 +51,145 @@ def run(self):
-                    tags += NS.node_context.tags
-                    NS.node_context.tags = list(set(tags))
-                    NS.node_context.save()
-                break
 
-        Event(
-            Message(
-                job_id=self.parameters['job_id'],
-                flow_id = self.parameters['flow_id'],
-                priority="info",
-                publisher=NS.publisher_id,
-                payload={"message": "Starting SDS install and config %s" % integration_id
-                     }
-            )
-        )
+                self.parameters.update({'create_mon_secret': True})
+                ceph_help.create_ceph(self.parameters)
+            else:
+                Event(
+                    Message(
+                        job_id=self.parameters['job_id'],
+                        flow_id = self.parameters['flow_id'],
+                        priority="info",
+                        publisher=NS.publisher_id,
+                        payload={"message": "Creating Gluster Storage Cluster %s" % integration_id
+                             }
+                    )
+                )
 
-        # SSH setup jobs finished above, now install sds bits and create cluster
-        if "ceph" in sds_name:
+                gluster_help.create_gluster(self.parameters)
+
+            # Wait till detected cluster in populated for nodes
             Event(
                 Message(
                     job_id=self.parameters['job_id'],
                     flow_id = self.parameters['flow_id'],
                     priority="info",
                     publisher=NS.publisher_id,
-                    payload={"message": "Creating Ceph Storage Cluster %s" % integration_id
+                    payload={"message": "SDS install and config completed, "
+                                        "Waiting for tendrl-node-agent to "
+                                        "detect newly installed sds details %s %s" % (integration_id,
+                                                                        self.parameters['Node[]'])
                          }
                 )
             )
 
-            self.parameters.update({'create_mon_secret': True})
-            ceph_help.create_ceph(self.parameters)
-        else:
+            while True:
+                gevent.sleep(3)
+                all_status = []
+                for node in self.parameters['Node[]']:
+                    try:
+                        NS._int.client.read("/nodes/%s/DetectedCluster/detected_cluster_id" %
+                                            node)
+                        all_status.append(True)
+                    except etcd.EtcdKeyNotFound:
+                        all_status.append(False)
+                if all_status:
+                    if all(all_status):
+                        break
+        except Exception as ex:
+            # releasing nodes if any exception came
+            create_cluster_utils.release_node_lock(
```

Comment:
For create and expand cluster we are create import job also... if i use finally at end of file then import cluster job is fired and then only finally is executed so import cluster will fail because nodes are already locked.